nbgrader/docs/source/user_guide/what_is_nbgrader.rst

```diff
@@ -0,0 +1,350 @@
+properly segmented into user accounts.  The **filesystem exchange**
+allows you to distribute the assignments, students to submit
+completed assignments, and release feedback.  The filesystem exchange
+is simple but effective.
+
+It is structured as:
+
+* ``outbound/``, containing assignments released to students.
+
+  * Organized as ``outbound/{assignment_id}/{notebook}.ipynb``.  Other
+    data files can be distributed along with the notebook.
+
+  * Files are copied from ``{course_dir}/release/`` to
+    ``{exchange_dir}/outbound/`` by the ``nbgrader release_assignment``
+    command.
+
+* ``inbound/``, containing assignments the student is submitting.  It
+  should be writeable, but not listable, by students (``-wx`` UNIX
+  permissions).
+
+  * Organized as
+    ``inbound/{studend_id}+{assignment_id}+{timestamp}+{random_string}/{notebook}.ipynb``.
+    One of these directories contains one submission of the
+    assignment.  The protection of them is within the random string,
+    which is described below.
+
+  * Files are copied from ``{exchange_dir}/inbound/`` to
+    ``{course_dir}/submitted/`` by the ``nbgrader fetch_assignment``
+    command.
+
+* ``feedback/``, containing feedback to the students.  This should
+  be traversable by students, but not listable (``--x``).  The
+  files inside should be readable (``r--``).
+
+  * Organized as ``feedback/{hash}.html``, where ``{hash}`` is a hash
+    of notebook contents and timestamp of submission.  It serves as a
+    key which is known to the student but not to other students, so
+    that they can identify their notebook and retrieve it.
+
+  * Files are copied from ``{course_dir}/feedback/`` to
+    ``{exchange_dir}/feedback/`` by the ``nbgrader release_feedback``
+    command.
+
+The filesystem exchange relies on certain UNIX filesystem semantics:
+if a user has write and execute permissions on a directory, they can
+create files inside of it but not list other files in there.  If each
+file has an unpredictable name (e.g. by a random string), students can
+not access each others files (this is used for submitting
+assignments).  Furthermore, they can access files they *do* know the
+names of (this is used for retrieving feedback).  In order for these
+assumptions to apply, students must access the  filesystem under
+different numeric UNIX user ids (UIDs).
+
+The filesystem exchange isn't limited to just one computer, though.
+**Network filesystems** exist and have the necessary UNIX semantics - in
+particular, the Network Filesystem (NFS).  This can easily be used to
+mount an exchange directory on multiple computers, so that students
+can be distributed among multiple computers within a cluster.
+However, this requires a consistent mapping to UIDs across the
+cluster.  This is not difficult to do, but if often not the way that
+"cloud stuff" works by default.
+
+The default filesystem exchange path is ``/srv/nbgraber/exchange``.
```

Comment:
Typo? `nbgraber` for `nbgrader` in `/srv/nbgraber/exchange`.

Reply:
Note: also seen at https://github.com/jupyter/nbgrader/pull/1000#issuecomment-427664752